extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -582,14 +588,29 @@ private Optional<SqlStatementResult> getStatementStatus(
     MSQControllerTask msqControllerTask = getMSQControllerTaskAndCheckPermission(queryId, authenticationResult, forAction);
     SqlStatementState sqlStatementState = SqlStatementResourceHelper.getSqlStatementState(statusPlus);
 
+    Supplier<Optional<MSQTaskReportPayload>> msqTaskReportPayloadSupplier = () -> {
+      try {
+        return Optional.ofNullable(SqlStatementResourceHelper.getPayload(
+            contactOverlord(overlordClient.taskReportAsMap(queryId), queryId)
+        ));
+      }
+      catch (DruidException e) {
+        if (e.getErrorCode().equals("notFound")) {
+          return Optional.empty();
```

Comment:
This happens initially when the overlord isn't yet aware of the query details. We shouldn't fail the API call if these fields aren't available to be fetched yet.

The not found error is handled in a similar way in the rest of the file. (`contactOverlord()` method throws it)

```java
private <T> T contactOverlord(final ListenableFuture<T> future, String queryId)
  {
    try {
      return FutureUtils.getUnchecked(future, true);
    }
    catch (RuntimeException e) {
      if (e.getCause() instanceof HttpResponseException) {
        HttpResponseException httpResponseException = (HttpResponseException) e.getCause();
        if (httpResponseException.getResponse() != null && httpResponseException.getResponse().getResponse().getStatus()
                                                                                .equals(HttpResponseStatus.NOT_FOUND)) {
          log.info(httpResponseException, "Query details not found for queryId [%s]", queryId);
          // since we get a 404, we mark the request as a NotFound. This code path is generally triggered when user passes a `queryId` which is not found in the overlord.
          throw queryNotFoundException(queryId);
        }
      }
      throw DruidException.forPersona(DruidException.Persona.DEVELOPER)
                          .ofCategory(DruidException.Category.UNCATEGORIZED)
                          .build("Unable to contact overlord " + e.getMessage());
    }
  }
```